app/api/rest_api.py

```diff
@@ -58,8 +61,22 @@ def __init__(self, services):
         data = dict(plugins=[p.display for p in plugins])
         return render_template('%s.html' % access[0].name, request, data)
 
+    @staticmethod
+    async def api_doc_page(request):
+        return render_template('apidocs.html', request, dict())
+
     """ API ENDPOINTS """
 
+    @swagger(summary='Core API Endpoint', description='Supports read, update, and delete operations',
+             requestBody={'content': 'application/json'},
```

Comment:
what i like:
* the tab on the home pages which breaks the docs in reasonable ways. 
* the separation of the API docs from our normal docs. 
* the GUI for swagger looks nice

what i don't like:
* the decorators seem pretty heavy/clunky in the rest_api module
* there doesn't seem to be a lot of value to the docs for a REST API user. it shows the available endpoints but nothing related to why i would use GET vs PUT vs POST for downloading files. and nothing breaking down the variations of options in the /api/rest endpoint.

Reply:
I agree. 

For the first problem we could pull some of these dictionaries into a different file (easy), or maybe make spec generator functions that would return these dictionaries.  We could also store this information completely outside of the python code in YAML, but I think that greatly increases the risk of the docs becoming outdated. 

For the second problem: 

1) (easy)  Update the description for rest_core , which should make things about PUT vs POST clearer. And I'll include the `index` parameter. 

2) (easy) Make some changes so that only POST/PUT/DELETE is shown for the rest_core endpoint. 

3) Down the road, I think it would make sense to split rest_core up into multiple routes (e.g. at least one per action).  I think it would be a little more intuitive for folks trying to figure out how things work and easier to document. 

4) We can use a 3rd party lib or tweak our objects a bit so that we can generate a schema that will include all the valid fields and types for our objects.  This actually wouldn't be too hard, but I think it is out of scope for this PR. 

I'll take a stab at doing the (easy) changes I mentioned above.